component/src/main/java/org/wso2/extension/siddhi/io/websocket/sink/websocketserver/WebSocketServerHandshakeListener.java

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.extension.siddhi.io.websocket.sink.websocketserver;
+
+import org.wso2.siddhi.core.exception.SiddhiAppRuntimeException;
+import org.wso2.transport.http.netty.contract.websocket.HandshakeListener;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import javax.websocket.Session;
+
+/**
+ * Future listener for WebSocket handshake.
+ */
+
+public class WebSocketServerHandshakeListener implements HandshakeListener {
+    private List<Session> sessionList = new CopyOnWriteArrayList<>();
```

Comment:
We don't need to create a new object since variable is assigned in constructor

Reply:
fixed in 39e7381